src/abstractdataframe/abstractdataframe.jl

```diff
@@ -1435,3 +1435,80 @@ function CategoricalArrays.categorical(df::AbstractDataFrame,
     end
     DataFrame(newcols, _names(df), copycols=false)
 end
+
+"""
+    flatten(df::AbstractDataFrame, col::Union{Integer, Symbol})
+
+When column `col` of data frame `df` has iterable elements that define `length` (for example
+a `Vector` of `Vector`s), return a `DataFrame` where each element of `col` is flattened, meaning 
+the column corresponding to `col` becomes a longer `Vector` where the original entries 
+are concatenated. Elements of row `i` of `df` other than `col` will be duplicated according to 
```

Comment:
There is one crucial thing to clarify in the docs before we merge this.

Here "duplicated" does not mean "copied". I think we should clarify this. Essentially this means that if we duplicate mutable objects the column will hold their aliases (which saves memory as a plus, but can be surprising if we do not explicitly mention this).

Do you agree that it is something that should be mentioned?

Reply:
You are right. We should either mention it or perform a copy. `flatten`ing and then modifying the original data frame could be a pain point for users.

Reply:
Yes - and I was not sure which approach is better. Let us wait for @nalimilan to comment on this (and we can merge the PR soon). Apart from this comment and a small comment below I think we are done. Thank you!

Reply:
Good catch. I don't think we can make copies preventively as many objects don't support `copy` and there's no trait to detect that. Just mentioning this should be enough. Maybe say "repeated" rather than "duplicated" for clarity.

Reply:
Yes - copy would have to be a `deepcopy` (sorry for not mentioning this). But let us go with what @nalimilan recommends - and just clarify in the documentation that we do not copy.